src/hooks/useRealmValidation.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback } from 'react'
+import { isEns, isPeerHealthy } from '../utils'
+import { fetchWorldContentAbout } from '../utils/worldContentApi'
+
+export interface RealmValidationResult {
+  isValid: boolean
+  validatedRealmOrWorld?: string
+  error?: string
+}
+
+/**
+ * Custom hook for realm validation logic
+ */
+export const useRealmValidation = () => {
+  const ensureProtocol = useCallback((input: string): string => {
+    const hasProtocol = /^[a-zA-Z][a-zA-Z\d+\-.]*:\/\//.test(input)
+    return hasProtocol ? input : `https://${input}`
+  }, [])
+
+  const checkRealmHealth = useCallback(async (realm: string): Promise<boolean> => {
```

Comment:
NIT:

```suggestion
  const checkRealmOrWorldHealth = useCallback(async (realmOrWorld: string): Promise<boolean> => {
```

or we can split to have two functions, as you wish